src/main/java/gov/cms/madie/terminology/service/VsacService.java

```diff
@@ -213,15 +213,27 @@ private String buildCodeSystemVersion(CqlCode cqlCode, CodeSystemEntry codeSyste
    *     errorCode 802 indicates Code not found.
    */
   private void buildVsacErrorMessage(CqlCode cqlCode, VsacCode vsacCode) {
-    int errorCode = Integer.parseInt(vsacCode.getErrors().getResultSet().get(0).getErrCode());
-    if (errorCode == 800 || errorCode == 801) {
-      cqlCode.getCodeSystem().setValid(false);
-      cqlCode
-          .getCodeSystem()
-          .setErrorMessage(vsacCode.getErrors().getResultSet().get(0).getErrDesc());
-    } else if (errorCode == 802) {
+    // Validation errors appear in an Errors object
+    if (vsacCode.getErrors() != null
+        && StringUtils.isNumeric(vsacCode.getErrors().getResultSet().get(0).getErrCode())) {
+      int errorCode = Integer.parseInt(vsacCode.getErrors().getResultSet().get(0).getErrCode());
+      if (errorCode == 800 || errorCode == 801) {
+        cqlCode.getCodeSystem().setValid(false);
+        cqlCode
+            .getCodeSystem()
+            .setErrorMessage(vsacCode.getErrors().getResultSet().get(0).getErrDesc());
+      } else if (errorCode == 802) {
+        cqlCode.setValid(false);
+        cqlCode.setErrorMessage(vsacCode.getErrors().getResultSet().get(0).getErrDesc());
+      }
+    // API Errors appear in the status field.
+    } else if (StringUtils.isNumeric(vsacCode.getStatus())) {
```

Comment:
so in this scenario, VSAC doesn’t provide an error message?

Reply:
Technically, it does: ```"error": "Bad Request",```

Reply:
ah..okay, thanks!